packages/token-dictionary/src/middleware.ts

```diff
@@ -71,6 +71,15 @@ export const addVirtualPalette: TokenMiddleware = {
     tokens.forEach((token) => {
       const { colorPalette } = token.extensions
       if (!colorPalette) return
+      const colorPaletteRoot = colorPalette.split('.').shift()
+      token.path
+        .slice(token.path.indexOf(colorPaletteRoot) + 1)
+        .reduce((acc, _, i, arr) => {
+          acc.push(arr.slice(0, i + 1).join('.'))
+          return acc
+        }, [] as string[])
+        .forEach((key) => keys.add(key))
```

Comment:
could you add a little comment above so we know with a glance what's happening here ?

Reply:
Ofc I'll do an refactoring and commenting phase, this is just a PoC for now.
I'm still worried about the performance and some memory leaks.

Reply:
For example, now `colorPalettes` have a lot of duplicate tokens
![image](https://github.com/chakra-ui/panda/assets/4128883/52b15faf-77b5-4d69-a351-8dc71e7c7474)

Reply:
Duplicates are now fixed 
![image](https://github.com/chakra-ui/panda/assets/4128883/80e5ce74-2d95-4f0c-857f-54fe7d4ab172)